metagpt/actions/extract_readme.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Module Description: This script defines the LearnReadMe class, which is an action to learn from the contents of
+    a README.md file.
+Author: mashenquan
+Date: 2024-3-20
+"""
+from pathlib import Path
+from typing import Optional
+
+from pydantic import Field
+
+from metagpt.actions import Action
+from metagpt.const import GRAPH_REPO_FILE_REPO
+from metagpt.schema import Message
+from metagpt.utils.common import aread
+from metagpt.utils.di_graph_repository import DiGraphRepository
+from metagpt.utils.graph_repository import GraphKeyword, GraphRepository
+
+
+class ExtractReadMe(Action):
+    """
+    An action to extract summary, installation, configuration, usages from the contents of a README.md file.
+
+    Attributes:
+        graph_db (Optional[GraphRepository]): A graph database repository.
+        install_to_path (Optional[str]): The path where the repository to install to.
+    """
+
+    graph_db: Optional[GraphRepository] = None
+    install_to_path: Optional[str] = Field(default="/TO/PATH")
+    _readme: Optional[str] = None
+    _filename: Optional[str] = None
+
+    async def run(self, with_messages=None, **kwargs):
+        """
+        Implementation of `Action`'s `run` method.
+
+        Args:
+            with_messages (Optional[Type]): An optional argument specifying messages to react to.
+        """
+        graph_repo_pathname = self.context.git_repo.workdir / GRAPH_REPO_FILE_REPO / self.context.git_repo.workdir.name
+        self.graph_db = await DiGraphRepository.load_from(str(graph_repo_pathname.with_suffix(".json")))
+        summary = await self._summarize()
+        await self.graph_db.insert(subject=self._filename, predicate=GraphKeyword.HAS_SUMMARY, object_=summary)
+        install = await self._extract_install()
+        await self.graph_db.insert(subject=self._filename, predicate=GraphKeyword.HAS_INSTALL, object_=install)
+        conf = await self._extract_configuration()
+        await self.graph_db.insert(subject=self._filename, predicate=GraphKeyword.HAS_CONFIG, object_=conf)
+        usage = await self._extract_usage()
+        await self.graph_db.insert(subject=self._filename, predicate=GraphKeyword.HAS_USAGE, object_=usage)
```

Comment:
why do we need to separate different parts of README. How are downstream tasks use such info?

Reply:
1. summary is used to restore user requirement.
2. usage is used to locate source code folder.
3. install is used to create install script in DI
4. configuration is used to create configuration script in DI